reference/api/search.md

```diff
@@ -735,3 +738,27 @@ Queries using `_geoPoint` will always include a `geoDistance` field containing t
 ```
 
 [You can read more about location-based sorting in our dedicated guide.](/learn/advanced/geosearch.md#sorting-results-with-geopoint)
+
+### Matching strategy
+
+**Parameter**: `matchingStrategy`
+**Expected value**: `last` or `all`
+**Default value**: `last`
+
+Defines the strategy used to match query terms in documents. `last` returns documents containing all the query words first. If there aren't enough documents to meet the requested `limit`, it will remove one query term at a time, starting from the end of the query.
+
+<CodeSamples id="search_parameter_guide_matching_strategy_1" />
+
+With the above code sample, Meilisearch will first return documents that contain all three words. If the results don't meet the requested `limit`, it will return documents with just `big fat` followed by just `big`.
+
+`all`, on the other hand, only returns documents that contain all the query words. Meilisearch will not match any more documents even if there aren't enough to meet the requested `limit`.
+
+Searching for `big fat liar` will only return documents that contain all three words.
+
+<CodeSamples id="search_parameter_guide_matching_strategy_2" />
+
+The above code sample only returns one result compared to 476 when the `matchingStrategy` was set to `last`.
+
+::: warning
+The number of search results may vary depending on changes to the `movies` dataset.
+:::
```

Comment:
I think we should remove mentions of specific numbers here instead of relying on the warning to dispel any confusion.

For example, we could replace this paragraph and the warning with a short text explaining that `all` will usually return fewer results than `last`.